crates/net/kademlia.rs

```diff
@@ -164,24 +167,41 @@ impl KademliaTable {
         }
 
         let peer = peer.unwrap();
-        peer.is_proven = false;
         peer.last_ping_hash = ping_hash;
         peer.last_ping = time_now_unix();
     }
 
+    pub fn update_peer_ping_with_revalidation(&mut self, node_id: H512, ping_hash: Option<H256>) {
+        let peer = self.get_by_node_id_mut(node_id);
+        if peer.is_none() {
+            return;
+        }
+
+        let peer = peer.unwrap();
+        peer.last_ping_hash = ping_hash;
+        peer.last_ping = time_now_unix();
+        peer.revalidation = Some(false);
+    }
+
     /// ## Returns
-    /// The peers whose pong back is past the since. Since is given in seconds and it is not an actual data
-    /// but a Duration.
+    /// The a vector of length of the provided `limit` of the peers who have the highest `last_ping` timestamp,
+    /// that is, those peers that were pinged least recently
     ///
-    /// ### Example
-    /// table.get_pinged_peers_since(24 * 60 * 60) // this would return the peers that haven't ponged for 24hs
-    pub fn get_pinged_peers_since(&mut self, since: u64) -> Vec<PeerData> {
+    /// ## Dev note:
+    /// This function could be improved
```

Comment:
Maybe a hint on how it can be improved for future reference.

Reply:
Updated!